docs/design.md

```diff
@@ -342,22 +342,22 @@ Transactions are managed by the client proxy (or gateway in SQL Azure
 parlance). Unlike in Spanner, writes are not buffered but are sent
 directly to all implicated ranges. This allows the transaction to abort
 quickly if it encounters a write conflict. The client proxy keeps track
-of all written keys in order to cleanup write intents upon transaction
-completion.
-
+of all written keys in order to post-process write intents upon completion.
 If a transaction is completed successfully, all intents are upgraded to
 committed. In the event a transaction is aborted, all written intents
-are deleted. The client proxy doesn’t guarantee it will cleanup intents;
```

Comment:
this asynchronous and best-effort nature of resolving intents is fairly important is now mentioned only later. I prefer having this up here to avoid stating one thing and then another below.